src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -58,10 +56,19 @@ public boolean searchName(String name) {
     }
 
     public void cleanBasket(ProductBasket product) {
-        for (int i = 0; i < products.length; i++) {
-            products[i] = null;
+        products.clear();
+    }
+
+    public void deleteProduct(String name) {
+        LinkedList<Product> delete = new LinkedList<>();
```

Comment:
```suggestion
        List<Product> delete = new LinkedList<>();
```
помним про полиморфизм